### PR TITLE
Add detection for CVE_2024_2194 & update webexploit base class to handel re.findall() returning list of tuple

### PR DESCRIPTION
**Pull request by ybadaoui-ostorlab**

### CODE REVIEW

File: agent/exploits/cve_2024_21762.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
         try:
             socket_wrapper.read(2048)
         except socket.error as exc:
-            if type(exc) == TimeoutError:
+            if type(exc) is TimeoutError:
```
Comment (3asm) [inline agent/exploits/cve_2024_21762.py:85]: Why aren't we doing except TimeoutError

**Review by 3asm — APPROVED** [review]

File: agent/exploits/cve_2024_2194.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+"""Agent Asteroid implementation for CVE-2024-2194"""
+
+import re
+from packaging import version
+from agent.exploits import webexploit
+from agent import exploits_registry
+from agent import definitions
```
Comment (rabsondev) [inline agent/exploits/cve_2024_2194.py:7]:
```suggestion
import re

from packaging import version

from agent.exploits import webexploit
from agent import exploits_registry
from agent import definitions
```

**Review by rabsondev — APPROVED** [review]

File: agent/exploits/webexploit.py [hunk 3]
```diff
@@ -101,6 +101,8 @@ def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
             if self.version_pattern is not None:
                 if (matched := self.version_pattern.findall(resp.text)) is not None:
                     for extracted_version in matched:
+                        if type(extracted_version) is tuple:
```
Comment (oussamaessaji) [inline agent/exploits/webexploit.py:104]: use isinstance instead

File: agent/exploits/cve_2024_21762.py (hunk 1 above)
Comment (ybadaoui-ostorlab) [reply agent/exploits/cve_2024_21762.py:85]: i am not sure, this is from an old CVE i modify it just to solve a mypy workflow error

File: agent/exploits/webexploit.py (hunk 3 above)
Comment (ybadaoui-ostorlab) [reply agent/exploits/webexploit.py]: fixed! thanks!

**Pull request merged**